bittensor/subtensor.py

```diff
@@ -817,6 +816,8 @@ def set_weights(
                     wait_for_finalization=wait_for_finalization,
                     prompt=prompt,
                 )
+                if not wait_for_finalization and not wait_for_inclusion:
```

Comment:
Would it not be better to break on `if success is True`?

Reply:
not sure if that will have the same desired output, but am open to it.